vnext/Shared/Modules/BlobModule.cpp

```diff
@@ -0,0 +1,222 @@
+         auto parts = jsArgAsArray(args, 0); // Array<Object>
+         auto blobId = jsArgAsString(args, 1);
+         vector<uint8_t> buffer{};
+
+         for (const auto &part : parts) {
+           auto type = part["type"];
+           if (type == "blob") {
+             auto blob = part["data"];
+             auto bufferPart = contentHandler->ResolveMessage(
+                 blob["blobId"].asString(), blob["offset"].asInt(), blob["size"].asInt());
+             buffer.reserve(buffer.size() + bufferPart.size());
+             buffer.insert(buffer.end(), bufferPart.begin(), bufferPart.end());
+           } else if (type == "string") {
+             auto data = part["data"].asString();
+             auto bufferPart = vector<uint8_t>(data.begin(), data.end());
+
+             buffer.reserve(buffer.size() + bufferPart.size());
+             buffer.insert(buffer.end(), bufferPart.begin(), bufferPart.end());
+           } else {
+             if (auto state = weakState.lock()) {
+               if (auto instance = state->Module->getInstance().lock()) {
+                 string message = "Invalid type for blob: " + type.asString();
+                 instance->callJSFunction(
+                     "RCTDeviceEventEmitter", "emit", dynamic::array("blobFailed", std::move(message)));
+               }
+             }
+             return;
+           }
+
+           contentHandler->StoreMessage(std::move(buffer), std::move(blobId));
+         }
+       }},
+
+      {"release",
+       [contentHandler = m_contentHandler](dynamic args) // blobId: string
+       {
+         auto blobId = jsArgAsString(args, 0);
+
+         contentHandler->RemoveMessage(std::move(blobId));
+       }}};
+}
+
+#pragma endregion CxxModule
+
+#pragma endregion BlobModule
+
+#pragma region BlobWebSocketModuleContentHandler
+
+#pragma region IWebSocketModuleContentHandler
+
+void BlobWebSocketModuleContentHandler::ProcessMessage(string &&message, dynamic &params) /*override*/ {
+  params["data"] = std::move(message);
+}
+
+void BlobWebSocketModuleContentHandler::ProcessMessage(vector<uint8_t> &&message, dynamic &params) /*override*/ {
+  auto blob = dynamic::object();
+  blob("offset", 0);
+  blob("size", message.size());
+
+  // substr(1, 36) strips curly braces from a GUID.
+  string blobId = winrt::to_string(winrt::to_hstring(GuidHelper::CreateNewGuid())).substr(1, 36);
+  StoreMessage(std::move(message), std::move(blobId));
+
+  params["data"] = std::move(blob);
+  params["type"] = "blob";
+}
+#pragma endregion IWebSocketModuleContentHandler
+
+void BlobWebSocketModuleContentHandler::Register(int64_t socketID) noexcept {
+  lock_guard<mutex> lock{m_socketIDsMutex};
+  m_socketIDs.insert(socketID);
+}
+
+void BlobWebSocketModuleContentHandler::Unregister(int64_t socketID) noexcept {
+  lock_guard<mutex> lock{m_socketIDsMutex};
+  if (m_socketIDs.find(socketID) != m_socketIDs.end())
+    m_socketIDs.erase(socketID);
```

Comment:
>socketID [](http://example.com/codeflow?start=22&length=8)

We effectively search twice. It is better to erase the iterator received from the find method.